### PR TITLE
New tcp

### PR DESCRIPTION
**Pull request by TristanLaan**

### CODE REVIEW

File: kernels/cclo/fw/sw_apps/ccl_offload_control/src/ccl_offload_control.h [hunk 1]
```diff
@@ -33,7 +33,7 @@
 //Datapath width (in bytes) for all streams everywhere in the design
 #define DATAPATH_WIDTH_BYTES 64
 
-//AXIS interfaces to/from MB
```
Comment (quetric) [inline kernels/cclo/fw/sw_apps/ccl_offload_control/src/ccl_offload_control.h:36]: no functional changes in this file that i can tell, please revert it to the version in dev so we shrink the size of the PR

File: test/hardware/Makefile [hunk 2]
```diff
@@ -29,7 +29,7 @@ FREQUENCY = 250
 CCLO_XO = ../../kernels/cclo/ccl_offload.xo
 CCLO_MB_DEBUG_LEVEL ?= 0
 
-VNX=xup_vitis_network_example
```
Comment (quetric) [inline test/hardware/Makefile:32]: no functional changes here either, please revert to dev

File: test/hardware/config/link_config_u250_axis3x.ini [hunk 3]
```diff
@@ -133,7 +133,7 @@ stream_connect=ccl_offload_2.m_axis_eth_tx_meta:network_krnl_2.s_axis_tcp_tx_met
 stream_connect=ccl_offload_2.m_axis_eth_tx_data:network_krnl_2.s_axis_tcp_tx_data
 
 # Connect tcp kernels in a loop
-# this is temporary, the connections will be broken and
```
Comment (quetric) [inline test/hardware/config/link_config_u250_axis3x.ini:136]: revert cosmetic changes

File: test/hardware/config/link_config_u280_axis3x.ini [hunk 4]
```diff
@@ -133,7 +133,7 @@ stream_connect=ccl_offload_2.m_axis_eth_tx_meta:network_krnl_2.s_axis_tcp_tx_met
 stream_connect=ccl_offload_2.m_axis_eth_tx_data:network_krnl_2.s_axis_tcp_tx_data
 
 # Connect tcp kernels in a loop
-# this is temporary, the connections will be broken and
```
Comment (quetric) [inline test/hardware/config/link_config_u280_axis3x.ini:136]: revert cosmetic changes

File: test/hardware/config/link_config_u55c_axis3x.ini [hunk 5]
```diff
@@ -134,7 +134,7 @@ stream_connect=ccl_offload_2.m_axis_eth_tx_meta:network_krnl_2.s_axis_tcp_tx_met
 stream_connect=ccl_offload_2.m_axis_eth_tx_data:network_krnl_2.s_axis_tcp_tx_data
 
 # Connect tcp kernels in a loop
-# this is temporary, the connections will be broken and
```
Comment (quetric) [inline test/hardware/config/link_config_u55c_axis3x.ini:137]: revert cosmetic changes

File: test/host/hls/test.cpp [hunk 6]
```diff
@@ -100,7 +100,7 @@ std::unique_ptr<ACCL::ACCL> test_vadd_put(options_t options) {
 
         src_bo.write(src);
         src_bo.sync(XCL_BO_SYNC_BO_TO_DEVICE);
-        auto run = vadd_ip(src_bo, dst_bo, options.count, (rank+1)%size, accl->get_communicator_addr(),
```
Comment (quetric) [inline test/host/hls/test.cpp:103]: revert cosmetic changes (spaces, empty lines added or deleted) here and everywhere. the PR is very large, we should condense it to the minimum viable change

File: driver/xrt/src/accl.cpp [hunk 7]
```diff
@@ -460,6 +479,8 @@ CCLO *ACCL::scatter(BaseBuffer &sendbuf,
                     communicatorId comm_id, bool from_fpga, bool to_fpga,
                     dataType compress_dtype, bool run_async,
                     std::vector<CCLO *> waitfor) {
+  check_tcp_ready();
```
Comment (quetric) [inline driver/xrt/src/accl.cpp:482]: it's unlikely we need a check for each call right? if we do, wouldn't it be more appropriate to put it in e.g. prepare_call?

**Review by quetric — CHANGES_REQUESTED** [review]

File: test/host/hls/test.cpp (hunk 6 above)
Comment (TristanLaan) [reply test/host/hls/test.cpp:103]: I'll revert the unchanged files, but I don't think it's worth the effort of trying to get back the trailing whitespace on files that have been changed.

**Review by quetric — APPROVED** [review]

**Pull request merged**